src/identity-provider.js

```diff
@@ -10,7 +11,7 @@ class IdentityProvider {
     this._keystore = keystore
   }
 
-  async createIdentity(id, identitySignerFn) {
+  async createIdentity(id, identitySignerFn, type) {
```

Comment:
Perhaps move the `type` arg before the signer function arg for clarity (id, type of id, signer of the id)?

Reply:
Sure the idea was so that it would work with the current ipfs-log but we can do a new PR with fixes but agreed its better to have type first